celery/canvas.py

```diff
@@ -502,18 +525,48 @@ def stamp(self, visitor=None, **headers):
         headers = headers.copy()
         if visitor is not None:
             visitor_headers = visitor.on_signature(self, **headers)
-            if "stamped_headers" not in visitor_headers:
+            if visitor_headers and "stamped_headers" not in visitor_headers:
                 visitor_headers["stamped_headers"] = list(visitor_headers.keys())
-            headers.update(visitor_headers)
+            _merge_dictionaries(headers, visitor_headers or {})
```

Comment:
When will `visitor_headers` be a falsy value?
528 checks if it isn't so why `or {}` is necessary?